core/src/main/scala/cats/syntax/monadError.scala

```diff
@@ -31,6 +38,11 @@ final class MonadErrorOps[F[_], E, A](private val fa: F[A]) extends AnyVal {
     F.adaptError(fa)(pf)
 }
 
+final class MonadErrorOps0[F[_], E, A](private val fa: F[A]) extends AnyVal {
+  def attemptTap[B](f: Either[E, A] => F[B])(implicit F: MonadError[F, E]): F[A] =
+    F.flatMap(F.attempt(fa))(ea => F.flatMap(f(ea))(_ => fa))
+}
```

Comment:
This will run `fa` twice, I'm not sure that that's what I'd expect 🤔

Reply:
Doh! Good catch. thanks and fixed